set up dev tooling&CI for juno_k3s

danielpodwysocki opened the pull request:
This sets up the initial dev tooling for testing&autodocs. I am already bumping it as a PR to make next reviews easier.

The docs here are conforming to a more ansible-style convention.

This is because the role in this repo is a component that the playbook repo will import (https://github.com/juno-fx/k8s-playbooks).

The more high-level usage docs will stil be following the Orion doc convention and will be added on the other repo.

This also sets up the CI and runs test against each Linux distirbution we support.
The test works like this: we spin up the EC2 instances needed to apply
our example playbook -> we apply it -> we note down the result in CI and
tear it down.

It cleans up even after failures, so we won't incur big
cost.


Details on the AWS-side conf are in: https://github.com/juno-fx/juno_k3s_private/issues/1

skeleturtle reviewed (changes requested):
I noticed just a couple of minor things. As I mentioned, this isn't my area of expertise and I am working on getting up to speed on it. Until then I will need to defer to Alex for a little more thorough PR review.

skeleturtle reviewed (approved):
This LGTM,  before merge, lets wait for Alex's review.

aldmbmtl reviewed (changes requested):
Just so I know, we aren't testing the Graviton/ARM's correct?

danielpodwysocki committed:
set up dev tooling&CI for juno_k3s

This sets up the initial dev tooling for testing&autodocs.
I am already bumping it as a PR to make next reviews easier.

The docs here are conforming to a more ansible-style convention.

This is because the role in this repo is a component that the playbook repo
will import (https://github.com/juno-fx/k8s-playbooks).

The more high-level usage docs will stil be following the Orion doc
convention and will be added on the other repo.

This also sets up the CI and runs test against each Linux distirbution we support.
The tests run sequentially for now - if this becomes a limitation, it
won't be too hard to parallelize them.

The test works like this: we spin up the EC2 instances needed to apply
our example playbook -> we apply it -> we note down the result in CI and
tear it down.

It cleans up even after failures, so we won't incur big
cost.

Details on the AWS-side conf are in: https://github.com/juno-fx/juno_k3s_private/issues/1

aldmbmtl reviewed (approved):
Looks great!
